models/gamesM.js

```diff
@@ -7,19 +7,134 @@ exports.selectCategories = () => {
   });
 };
 
-exports.selectReviews = () => {
-  return pool
-    .query(
-      `SELECT title, category, designer, owner, review_img_url, reviews.created_at, reviews.review_id, reviews.votes, CAST(COUNT(comments.comment_id) AS INTEGER) as comment_count FROM reviews
-      LEFT JOIN comments
-      ON reviews.review_id = comments.review_id
-      GROUP BY reviews.review_id
-      ORDER BY reviews.created_at DESC
-      ;`
-    )
-    .then((result) => {
-      return result.rows;
+// exports.selectReviews = (category, sort_by, order) => {
+//   return pool
+//     .query(
+//       `SELECT
+
+//       title,
+//       category,
+//       designer,
+//       owner,
+//       review_img_url,
+//       reviews.created_at,
+//       reviews.review_id,
+//       reviews.votes,
+//       CAST(COUNT(comments.comment_id) AS INTEGER) as comment_count FROM reviews
+
+//       LEFT JOIN comments
+
+//         ON reviews.review_id = comments.review_id
+
+//       ${category ? `WHERE reviews.category=$1` : ``}
+//       GROUP BY reviews.review_id
+//       ORDER BY reviews.${sort_by} ${order}
+//     ;`,
+//       category ? [category] : []
+//     )
+//     .then((result) => {
+//       if (result.rows.length > 0) {
+//         return result.rows;
+//       } else if (category) {
+//         return [];
+//       } else {
+//         throw { status: 404, msg: "No reviews found" };
+//       }
+//     });
+// };
+
+exports.selectReviews = (sort_by = "created_at", order = "desc", category) => {
+  const validSortBy = [
+    "created_at",
+    "owner",
+    "category",
+    "review_id",
+    "votes",
+    "comment_count",
+    "title",
+    "designer",
+  ];
+
+  const validCategory = [
+    "euro game",
+    "dexterity",
+    "social deduction",
+    "children's games"
+  ]
+
+  let querySoFar = `SELECT 
+
+  title, 
+  category, 
+  designer, 
+  owner, 
+  review_img_url, 
+  reviews.created_at, 
+  reviews.review_id, 
+  reviews.votes, 
+  CAST(COUNT(comments.comment_id) AS INTEGER) as comment_count FROM reviews
+
+  LEFT JOIN comments 
+  ON reviews.review_id = comments.review_id`;
+
+  if (category && !validCategory.includes(category)) {
```

Comment:
You could get this check out of the way at the top - just so your started logic on the query string isn't abandoned